pypeit/scripts/ql_keck_lris.py

```diff
@@ -0,0 +1,475 @@
+       Files at B position for this offeset
+    caliBrate (object):
+       CaliBrate object
+    spectrograph (object):
+       spectrograph object
+    det (int):
+       Detector number
+    parset (parsect object)
+       Parset
+    show (bool, optional):
+       Show 2d reduction outputs. Default=False
+    std_trace (string, optional)
+       Trace for standard star. Default=None
+
+    Returns
+    -------
+    spec2DObj_A, spec2DObj_B
+
+    spec2DObj_A (object, Spec2D):
+       Spec2d Object for extraction at A position
+    spec2DObj_B (object, Spec2D)
+       Spec2d Object for extraction at B position
+
+    """
+
+    # Build Science image
+    sciImg = buildimage.buildimage_fromlist(
+        spectrograph, det, parset['scienceframe'], list(files),  bias=caliBrate.msbias,
+        bpm=caliBrate.msbpm, slits=caliBrate.slits, ignore_saturation=False)
+
+    # Instantiate Reduce object
+    # Required for pypeline specific object
+    # At instantiaton, the fullmask in self.sciImg is modified
+    redux = reduce.Reduce.get_instance(sciImg, spectrograph, parset, caliBrate, 'science', ir_redux=True, show=show, det=det)
+
+    # skymodel, objmodel, ivarmodel, outmask, sobjs, scaleimg, waveimg, tilts = redux.run(
+    #     std_trace=std_trace, return_negative=True, show_peaks=show)
+
+    global_sky, sobjs_obj, skymask = redux.run_objfind(std_trace=std_trace, show_peaks=show)
+    skymodel, objmodel, ivarmodel, outmask, sobjs, scaleimg, waveimg, tilts = redux.run_extraction(
+        global_sky, sobjs_obj, skymask)
+
+    # TODO -- Do this upstream
+    # Tack on detector
+    for sobj in sobjs:
+        sobj.DETECTOR = sciImg.detector
+
+    # Construct table of spectral flexure
+    spec_flex_table = Table()
+    spec_flex_table['spat_id'] = caliBrate.slits.spat_id
+    spec_flex_table['sci_spec_flexure'] = redux.slitshift
+
+    # Construct the Spec2DObj with the positive image
+    spec2DObj = spec2dobj.Spec2DObj(sciimg=sciImg.image,
+                                    ivarraw=sciImg.ivar,
+                                    skymodel=skymodel,
+                                    objmodel=objmodel,
+                                    ivarmodel=ivarmodel,
+                                    scaleimg=scaleimg,
+                                    waveimg=waveimg,
+                                    bpmmask=outmask,
+                                    detector=sciImg.detector,
+                                    sci_spat_flexure=sciImg.spat_flexure,
+                                    sci_spec_flexure=spec_flex_table,
+                                    vel_corr=None,
+                                    vel_type=parset['calibrations']['wavelengths']['refframe'],
+                                    tilts=tilts,
+                                    slits=copy.deepcopy(caliBrate.slits))
+    spec2DObj.process_steps = sciImg.process_steps
+    return spec2DObj
+
+
+class QLKECKLRIS(scriptbase.ScriptBase):
```

Comment:
What does this method do that `QLMOS` does not?

Reply:
The QLMOS scripts can only reduce a single exposure. This script can stack many observations of the same slit and accommodates dithering. It is clear that having a bunch of versions of the same code that are instrument specific is a poor setup, which I acknowledge. However, the current quicklook needs to be completely refactored in my opinion to allow users to stack multiple observations. Looking at a single MOSFIRE slit in A-B mode for a 200s exposure is pretty useless if you are e.g. integrating for hours. The same goes for LRIS and DEIMOS, which is why this new quicklook approach has been developed. Right now I'd like to just get these in as placeholders until we can dedicate the energy to refactor the quicklook adopting this approach.